src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -214,12 +206,16 @@ impl ValueRowDeserializer for Deserializer {
         let data_start_idx = offsets_start_idx + datum_num * offset_bytes;
         let offsets = &encoded_bytes[offsets_start_idx..data_start_idx];
         let data = &encoded_bytes[data_start_idx..];
-        let mut datums: Vec<Option<Datum>> = vec![None; self.schema.len()];
-        let mut contained_indices = BTreeSet::new();
+
+        // initialize datums with default values
+        let mut datums: Vec<Datum> = vec![None; self.schema.len()];
+        for (i, datum) in &self.default_column_values {
+            datums[*i].clone_from(datum);
+        }
```

Comment:
Part of the performance gain is because I initialized the default values no matter whether it's overwritten. Thus, we simply return `Ok(datums)`.

Reply:
This step of processing is not specific to the current row. I'm wondering if preprocessing it into `self.default_row`, then directly cloning the full row can perform even better. 🤔